src/app.ts

```diff
@@ -1,26 +1,116 @@
+  .option(
+    "-n, --file-pair-output-limit",
+    "Specifies how many matching file pairs are shown in the result. All pairs are " +
+      "shown then this option is omitted.",
+  )
+  .option(
+    "-s, --minimum-lines-shortest <integer>",
+    "The minimum amount of lines in the shortest code fragment in a comparison before it is shown",
+    1,
+  )
+  .option(
+    "-S, --minimum-lines-longest <integer>",
+    "The minimum amount of lines in the longest code fragment in a comparison before it is shown",
+    1,
+  )
+  .option(
+    "-g, --maximum-gap-size <integer>",
+    "If two fragments are close to each other, they will be merged into a single fragment if the gap between them is " +
+      "smaller than the given number of lines." +
+      0,
+  )
+  .arguments("<locations...>")
+  .action(filesArgs => {
+    locations = filesArgs;
+  });
+
+program.on("--help", () => {
+  console.log(`
+Examples:
+Gives dolos all the files in the current directory and tells dolos that they are in javascript
+  $ dolos -l javascript *.js
+
+Specifies the gap size.
+  $ dolos *.js -g 0");
+  [[[0, 2], [9, 11]], [[4, 5], [13, 14]]]");
+  $ dolos *.js -g 1");
+  [[[0, 5], [9, 14]]]");
+    `);
+});
+
+program.parse(process.argv);
 
 (async () => {
-  const tokenizer = new CodeTokenizer("python");
-  const comparison = new Comparison(tokenizer);
-  // let path = 'samples/python/snowball-numbers/';
-
-  // let files = fs.readdirSync(path);
-  // await comparison.addFiles([path + files[0], path + files[1]]);
-
-  // let arr = [];
-  // for (const key of Array(files.length -3 ).keys()) {
-  //   arr.push(path + files[key + 2]);
-  // }
-  // const result = await comparison.compareFiles(arr);
-
-  await comparison.addFiles(["samples/js/sample.js"]);
-  const result = await comparison.compareFiles([
-    "samples/js/copied_function.js",
-    "samples/js/another_copied_function.js",
-  ]);
-  const summary = new Summary(result);
-  console.log(summary.toString());
+  const tokenizer = new CodeTokenizer(program.language);
+
+  if (locations.length < 2) {
+    console.error("Need at least two locations");
+    program.outputHelp();
+    process.exit(1);
+  }
+
+  // Compare all the file with each other.
+  const comparison = new Comparison(tokenizer, {
+    filterFragmentByPercentage: program.maximumFragmentCount === undefined,
+    maxFragment:
+      program.maximumFragmentCount === undefined
+        ? program.maximumFragmentPercentage
+        : program.maximumFragmentCount,
+  });
+
+  // Compare the base file with all the other files when there is a base file.
+  if (program.base) {
+    comparison.addFileToFilterList(program.base);
+  }
+
+  await comparison.addFiles(locations);
+  const matchesPerFile: Map<string, Matches<number>> = await comparison.compareFiles(locations);
+
+  const filterOptions: FilterOptions = {
+    fragmentOutputLimit: program.filePairOutputLimit,
```

Comment:
This was renamed to filePairOutputLimit, but it seems the behaviour stayed the same? If I'm correct this limits the number of fragments that are reported per file pair?

What is should be is the number of file pairs that are reported. For example, when you have 100 files/users, you have 10000 filepairs. You only want to report on the top n pairs.

Reply:
Ah sorry I misunderstood you last time. I will change the implementation.